### PR TITLE
Part 2 of the lab.

### PR DESCRIPTION
**Pull request by VictorVillacis**
Please review changes to main, especially with regards to command.

### CODE REVIEW

File: TinyOS/Assembler1.s [hunk 1]
```diff
@@ -64,7 +64,7 @@ Mega328P_Init:
 		sts	ADCSRA,r16		//stores 0x87 into ADC configuration SRAM  location
 		ldi r16,0x40		//Loads 0x40 into r16
 		sts ADMUX,r16		//Stores the value of r16 into the ADC multiplexer selection memory space in SRAM
-		ldi r16,0			//Loads 0 into r16
+		ldi r16,7			//Loads 0 into r16
```
Comment (kafous02) [inline TinyOS/Assembler1.s:67]: Can I get an explanation of this value?

File: TinyOS/Assembler1.s [hunk 2]
```diff
@@ -167,17 +174,19 @@ UART_Put:
 
 .global ADC_Get
 ADC_Get:
-		ldi		r16,0xC7			//Load 199 into r16
-		sts		ADCSRA,r16			//Move 199 from r16 to ADC Control and Status Register A
+		ldi		r16,0xE7			//Load 231 into r16
```
Comment (kafous02) [inline TinyOS/Assembler1.s:178]: And an explanation for this value/address?

File: TinyOS/main.c [hunk 3]
```diff
@@ -26,16 +27,25 @@ void LCD_Write_Command(void);
...
+void scrollingLCD(char*);
+void LCD_Delay(void);
 
 unsigned char ASCII;			//shared I/O variable with Assembly
 unsigned char DATA;				//shared internal variable with Assembly
 char HADC;						//shared ADC variable with Assembly
 char LADC;						//shared ADC variable with Assembly
 
-char volts[5];					//string buffer for ADC output
+char lastTemperature[5] = "     ";
```
Comment (kafous02) [inline TinyOS/main.c]: Not sure what this is doing, empty space?

File: TinyOS/main.c [hunk 4]
```diff
@@ -80,7 +90,12 @@ void LCD(void)						//Lite LCD demo
 	LCD_Write_Command();
 	DATA = 0x0f;					//Student Comment Here
 	LCD_Write_Command();
-	LCD_Puts("Hello ECE412!");
+	
+	char message[50] = "Group 4 is the best group in the world. ";
```
Comment (kafous02) [inline TinyOS/main.c]: U rite

File: TinyOS/main.c [hunk 5]
```diff
@@ -89,25 +104,121 @@ void LCD(void)						//Lite LCD demo
...
-	if (volts[0x2] == 58)
-	{
-		volts[0x2] = 48;
-		volts[0x0]++;
-	}
-	UART_Puts(volts);
-	UART_Puts(MS6);
+	//int stayInLoop = 1;
+
+	//while(stayInLoop){
```
Comment (kafous02) [inline TinyOS/main.c]: Can we get rid of these comments?

File: TinyOS/main.c [hunk 6]
```diff
@@ -89,25 +104,121 @@ void LCD(void)						//Lite LCD demo
...
+		char f2 = fahrenheit + '0';
+
+		if(f0 == '0'){
+			f0 = ' ';
+		}
+	
+		temperature[0] = f0;
+		temperature[1] = f1;
+		temperature[2] = f2;
+		temperature[3] = 167;
```
Comment (kafous02) [inline TinyOS/main.c]: What is this number?

**Review by kafous02 — COMMENTED** [review]
Mainly questions for my own understanding. There are a few formatting things like extra white space and some weird indenting, but that can be fixed quickly with a ninja commit on master. 

Also if you use a number that isn't really explained, like just a value that you needed as an offset or something, please put a comment next to it to say what that number is for. That way I don't ask a bunch of dumb unnecessary questions.

Good work you all!

File: TinyOS/main.c (hunk 3 above)
Comment (VictorVillacis) [reply TinyOS/main.c]: It's used to store the previous temperature. The display is only updated when there is a difference between the last temperature and the current temperature, and then the last temperature is updated.

File: TinyOS/Assembler1.s (hunk 2 above)
Comment (VictorVillacis) [reply TinyOS/Assembler1.s:178]: Loading the proper flags into the ADCSRA

File: TinyOS/Assembler1.s (hunk 1 above)
Comment (VictorVillacis) [reply TinyOS/Assembler1.s:67]: Loading the proper flags into the ADCSRB for FreeRunning Mode

File: TinyOS/main.c (hunk 6 above)
Comment (VictorVillacis) [reply TinyOS/main.c]: The degree symbol in ASCII

**Comment by kafous02** [comment]
Looks good, merging!

**Pull request merged**